lib/geoengineer/resources/aws/sqs/aws_sqs_queue_policy.rb

```diff
@@ -0,0 +1,51 @@
+########################################################################
+# AwsSqsQueuePolicy is the +aws_sqs_queue_policy+ terrform resource,
+#
+# {https://www.terraform.io/docs/providers/aws/r/sqs_queue_policy.html Terraform Docs}
+########################################################################
+class GeoEngineer::Resources::AwsSqsQueuePolicy < GeoEngineer::Resource
+  validate -> { validate_required_attributes([:queue_url, :policy]) }
+
+  after :initialize, -> {
+    _terraform_id -> {
+      queue_url
+    }
+  }
+
+  def to_terraform_state
+    tfstate = super
+    tfstate[:primary][:attributes] = {
+      "queue_url" => queue_url,
+      "policy" => policy
+    }
+    tfstate
+  end
+
+  def support_tags?
+    false
+  end
+
+  def self._fetch_remote_resources(provider)
+    queue_policies = []
+
+    AwsClients.sqs(provider).list_queues['queue_urls'].map do |queue_url|
```

Comment:
There's two things about the style of this line:

1. All the properties have accessor methods and the only place where it is read as element is queue_urls, so the better form would be `AwsClients.sqs(provider).list_queues.queue_urls` to maintain consistency with the rest of the code.
2. While iterating with `map` works, in this case you're not mutating the Array returned by `queue_urls`, you're actually appending to `queue_policies`, so `each` is the better iterator. Using `map` is a bad form in this context. I've suggested to explicitly declare and append to an Array as the code is more readable than the map / compact pattern which is a bit opaque for people not used to Ruby. While the performance impact is negligible (for this use case), maintaining consistency with the rest of the code is one of the goals.